includes/shell/environment.hpp

```diff
@@ -0,0 +1,36 @@
+#pragma once
+
+#include "shell/utility/types.hpp"
+
+namespace shell {
+
+namespace Environment {
+
+	/*!
+	 * @brief
+	 * @return
+	 */
+	char* const* getEnvironmentVariables();
+
+	/*!
+	 * @brief Adds variable to the environment variables, will overwrite if name already exits.
+	 * @param variable In the form `name=value`
+	 */
+	void addEnvironmentVariable(const string& variable);
```

Comment:
I think we want these to be separate arguments, so we have `name` and `value` separated

Reply:
actually.. maybe not, because we don't separate the token on `=` anyways.
I thought the logic of constructing the `key=value` string shouldn't be placed on the caller and instead be encapsulated in here

But we never separate the `key=value` string we receive anyways
`key = value` is not valid to begin with

Reply:
yup, thats what i thought too :)